app/qml/form/editors/MMFormPhotoViewer.qml

```diff
@@ -53,50 +47,58 @@ MMBaseInput {
 
   state: "notSet"
 
-  onContentClicked: {
-    if ( photo.status === Image.Ready ) {
-      previewLoader.active = true
-      previewLoader.focus = true
-    }
-  }
+  inputContent: Rectangle {
+    width: parent.width
+    height: __style.row160
+
+    color: __style.polarColor
+    radius: __style.radius20
 
-  content: Item {
-    MMPhoto {
+    MMComponents.MMPhoto {
       id: photo
 
-      width: root.width
-      height: root.contentItemHeight
+      width: parent.width
+      height: parent.height
+
       visible: root.state !== "notSet"
-      photoUrl: root.photoUrl
 
+      photoUrl: root.photoUrl
       fillMode: Image.PreserveAspectCrop
 
       MouseArea {
         anchors.fill: parent
-        onClicked: root.contentClicked()
+        onClicked: {
+          if ( photo.status === Image.Ready ) {
+            previewLoader.active = true
+            previewLoader.focus = true
+          }
+        }
       }
 
-      MMRoundButton {
+      MMComponents.MMRoundButton {
         anchors {
           right: parent.right
           bottom: parent.bottom
-          rightMargin: 10 * __dp
-          bottomMargin: 10 * __dp
+          rightMargin: __style.margin10
+          bottomMargin: __style.margin10
         }
 
         bgndColor: __style.negativeColor
         iconSource: __style.deleteIcon
         iconColor: __style.grapeColor
-        visible: root.allowEditing && root.state !== "notSet"
+
+        visible: root.editState === "enabled" && root.state !== "notSet"
```

Comment:
is "notSet" a leftover? It's not in the base class but is used elsewhere in this file too

Reply:
It is a part of this `MMFormPhotoViewer` file - but good catch, maybe it should not be set as a state of base class. I will add it to a sub-state group instead 👍🏻